services/housekeeper/housekeeper.go

```diff
@@ -164,13 +225,46 @@ func (hk *Housekeeper) updateKnownValidators() {
 
 	for _, v := range validators {
 		pubkey := types.PubkeyHex(v.Validator.Pubkey)
-		err = hk.redis.SetKnownValidator(pubkey, v.Index)
+		err := hk.redis.SetKnownValidator(pubkey, v.Index)
 		if err != nil {
 			log.WithError(err).WithField("pubkey", pubkey).Fatal("failed to set known validator in Redis")
 		}
 	}
 }
 
+func (hk *Housekeeper) fetchValidators() map[types.PubkeyHex]beaconclient.ValidatorResponseEntry {
+	// return the first successful beacon node response
+	var result map[types.PubkeyHex]beaconclient.ValidatorResponseEntry
+	clients := hk.getBeaconClientsByLastResponse()
+
+	var mu sync.Mutex
+	for i, client := range clients {
+		log := hk.log.WithField("uri", client.GetURI())
+		log.Debug("fetching validators")
+
+		mu.Lock()
```

Comment:
the lock seems not needed?

Reply:
should probably use atomic to modify the headslot here, yes. 

but the lock you are using is a local lock, and only used in this loop, so the mutex is not used in any concurrency way 🤔